### PR TITLE
qa: mininode: Clearer error message on invalid magic bytes

### PR DESCRIPTION
**Pull request by maflcko**
Old message:
```
ValueError: got garbage b'\xfa\xbf\xb5\xdafeefilter\x00\x00\x00\x08\x00\x00\x00\xe8\x0f\xd1\x9f\xe8\x03\x00\x00\x00\x00\x00\x00'
```

New message:
```
ValueError: magic bytes mismatch: b'\x00\x11"2' != b'\xfa\xbf\xb5\xdapong\x00\x00\x00\x00\x00\x00\x00\x00\x08\x00\x00\x00\x97\xe6\x04\xd2\xff\x00\x00\x00\x00\x00\x00\x00'
```

### CODE REVIEW

File: test/functional/test_framework/mininode.py [hunk 1]
```diff
@@ -171,7 +171,7 @@ def _on_data(self):
                 if len(self.recvbuf) < 4:
                     return
                 if self.recvbuf[:4] != self.magic_bytes:
-                    raise ValueError("got garbage %s" % repr(self.recvbuf))
+                    raise ValueError("magic bytes mismatch: {} != {}".format(repr(self.magic_bytes), repr(self.recvbuf)))
```
Comment (practicalswift) [inline test/functional/test_framework/mininode.py]:
```suggestion
                    raise ValueError("magic bytes mismatch: {} != {}".format(repr(self.magic_bytes), repr(self.recvbuf[:4])))
```

**Review by practicalswift — CHANGES_REQUESTED** [review]
Concept ACK

File: test/functional/test_framework/mininode.py (hunk 1 above)
Comment (MarcoFalke) [reply test/functional/test_framework/mininode.py]: Would still be nice to see the message type

Comment (practicalswift) [reply test/functional/test_framework/mininode.py]: That's a good point. Reverting my suggestion :-)

Comment (practicalswift) [inline test/functional/test_framework/mininode.py]:
```suggestion
                    raise ValueError("magic bytes mismatch: received {} does not start with the expected {}".format(self.recvbuf, self.magic_bytes))
```

**Review by practicalswift — CHANGES_REQUESTED** [review]

File: test/functional/test_framework/mininode.py (hunk 1 above)
Comment (promag) [reply test/functional/test_framework/mininode.py]: Was going to suggest something like this.

**Comment by laanwj** [comment]
Much better message, utACK fab0a68aa25191a532f26cf8e227f456b853d8f1

**Pull request merged**